### DISCUSSION TITLE
Rstudio Quarto render [no-watch-inputs --no-browse]

### DISCUSSION THREAD

**Discussion by BBQ71** [quarto-dev/quarto-cli#11023] [Q&A] [answered]
### Description

when I want to render a document, it will wanring 
`==> quarto preview test.qmd --to html --no-watch-inputs --no-browse`
and nothing appears in the viewer on the right 
How to solve this problem？

my Rstudio version：RStudio 2024.09.0 
R bsae version：4.4.1
quarto version：1.5.57 Date Aug 30,2024

Also，I have installed.packages("quarto")

**Comment by mcanouil** [comment]
Because you're issue seems to be RStudio IDE rather than Quarto CLI, I suggest you update your version of RStudio which is nearly one year old.

> when I want to render my code

What does that mean? What do you do to render a document?

> Also，I have installed.packages("quarto")

This is unrelated. As explained in the R package documentation, "quarto" R package is only a wrapper that allows to call Quarto CLI.

**Reply by BBQ71** [reply]
> > when I want to render my code

> What does that mean? What do you do to render a document?

Thank you! Yes, I mean render a document.
I've updated Rstudio to version 2024.09.0, but it still not work, really confuses me.

**Reply by mcanouil** [reply]
> Thank you! Yes, I mean render a document.

It does not tell me what you do to render a document.

Are you using Rstudio interface? Are you running Quarto from the command line?
Please describe what you are doing in order for us to help you.

**Reply by BBQ71** [reply]
![image](https://github.com/user-attachments/assets/4739a989-70c1-4001-a596-fd79a3576ae3)
> It does not tell me what you do to render a document.
Maybe I can't understand you very exactly, I guess I did what you siad.

By the way ,I found it also warned me

```
Error running filter C:/Users/71/AppData/Local/Programs/Quarto/share/filters/main.lua:
[string "..."]:295: module 'modules/astshortcode' not found:
```

**Reply by mcanouil** [reply]
@BBQ71 What am I supposed to guess from the screenshot? Are you clicking on a button? are you running `quarto preview file.qmd` in a terminal?
I am not an RStudio user or at least I did not use it in the last 4/5 years.

Quarto is a command line tools, meaning it runs mainly in a terminal.
IDE such as RStudio integrates Quarto CLI by "hiding" it behind buttons, by doing so Quarto CLI could not behave the same way because of IDE constraints.

From my years on RStudio, since the command appears in "background jobs" panel (and I think you unlikely did that yourself, sorry if I underestimate you by trying to guess), I would bet you are using the "render" (or "preview" button) from the IDE.
If that is correct, could you run on the "terminal" panel: `quarto check`?

**Reply by cderv** [reply]
> Error running filter C:/Users/71/AppData/Local/Programs/Quarto/share/filters/main.lua:
[string "..."]:295: module 'modules/astshortcode' not found:

This is suspicious. It seems you are using a version of Quarto install on your system at `C:/Users/71/AppData/Local/Programs/Quarto`. You should check this version, especially if you are not aware of it. It could be an old one, or a wrongly installed one. 

RStudio bundles quarto too and you could try uninstalling your system version and see it the error still happens. 

Anyhow, I would look into this as Clicking render is working with last Quarto version and latest RStudio. I can't reproduce.

**Reply by BBQ71** [reply, marked as answer]
Thank you, I've already solved it. I set my computer's language to English, I changed it in all computer environment include uesr name,file location and so on. And then it worked well.
